### PR TITLE
Search/fix folder sort

### PR DESCRIPTION
**Pull request by Clarity-89**
**What this PR does / why we need it**:
- Fetch only dashboards from backend for List layout
- Move layout handling inside query reducer/hook
- Fix 'min-width' style prop for Select
- Fix Starred filter toggle

**Which issue(s) this PR fixes**:

Fixes ##23861

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/search/components/ActionRow.tsx [hunk 1]
```diff
@@ -37,8 +39,10 @@ export const ActionRow: FC<Props> = ({
 
   return (
     <div className={styles.actionRow}>
-      <HorizontalGroup spacing="md">
-        {!hideLayout ? <RadioButtonGroup options={layoutOptions} onChange={onLayoutChange} value={layout} /> : null}
+      <HorizontalGroup spacing="md" width="100%">
```
Comment (torkelo) [inline public/app/features/search/components/ActionRow.tsx:42]: this should not be width 100% ? will squeeze the tag select and not allow it to expan d

**Review by torkelo — APPROVED** [review]
Looks good, but not sure about the width: 100% on the layout HorizontalGroup

File: public/app/features/search/components/ActionRow.tsx (hunk 1 above)
Comment (Clarity-89) [reply public/app/features/search/components/ActionRow.tsx:42]: hmm, maybe I've applied a wrong change during master merge.

Comment (Clarity-89) [reply public/app/features/search/components/ActionRow.tsx:42]: It actually has to be `width="auto"` for tag filter to expand properly.

**Pull request merged**